Makefile

```diff
@@ -0,0 +1,158 @@
+# setup - Sets up the entire development environment (pyenv and dependencies)
+# clean_docs - Clean the documentation folder
+# clean - Clean any generated files (including documentation)
+# open_docs - Open any docs generated with "make docs"
+# docs - Generated sphinx docs
+# validate - Run code validation
+# test - Run tests
+# run - Run any services for local development (databases, CSS compiliation, airflow, etc)
+# version - Show the version of the package
+
+OS = $(shell uname -s)
+
+MODULE_NAME=temporal_sqlalchemy
+
+ifdef CIRCLECI
+TEST_COMMAND=pytest --junitxml=$(TEST_REPORTS)/junit.xml
+# Use CircleCIs version
+PYTHON_VERSION=
+# Dont log pip install output since it can print the private repo url
+PIP_INSTALL_CMD=pip install -q
+# Do local installs without editable mode because of issues with CircleCI's venv
+PIP_LOCAL_INSTALL_CMD=pip install -q .
+else
+TEST_COMMAND=pytest
+DEV_PYTHON_VERSION=3.6.2
+PIP_INSTALL_CMD=pip install
+PIP_LOCAL_INSTALL_CMD=pip install -e .
+endif
+
+
+# Print usage of main targets when user types "make" or "make help"
+help:
+	@echo "Please choose one of the following targets: \n"\
+	      "    setup: Setup your development environment and install dependencies\n"\
+	      "    test: Run tests\n"\
+	      "    validate: Validate code and documentation\n"\
+	      "    docs: Build Sphinx documentation\n"\
+	      "    open_docs: Open built documentation\n"\
+	      "\n"\
+	      "View the Makefile for more documentation about all of the available commands"
+	@exit 2
+
+
+# Sets up pyenv and the virtualenv that is managed by pyenv
+.PHONY: pyenv
+pyenv:
+ifeq (${OS}, Darwin)
+	brew install pyenv pyenv-virtualenv 2> /dev/null || true
+# Ensure we remain up to date with pyenv so that new python versions are available for installation
+	brew upgrade pyenv pyenv-virtualenv 2> /dev/null || true
+endif
+
+# Only make the virtualenv if it doesnt exist
+	@[ ! -e ~/.pyenv/versions/${MODULE_NAME} ] && pyenv virtualenv ${DEV_PYTHON_VERSION} ${MODULE_NAME} || :
+	pyenv local ${MODULE_NAME}
+ifdef DEV_PYTHON_VERSION
+# If the Python used for development has been upgraded, remove the virtualenv and recreate it
+	@[ `python --version | cut -f2 -d' '` != ${DEV_PYTHON_VERSION} ] && echo "Python has been upgraded since last setup. Recreating virtualenv" && pyenv uninstall -f ${MODULE_NAME} && pyenv virtualenv ${DEV_PYTHON_VERSION} ${MODULE_NAME} || :
+endif
+
+	# Install all supported Python versions. There are more recent patch releases
+	# for most of these but CircleCI doesn't have them preinstalled. Installing a
+	# version of Python that isn't preinstalled slows down the build significantly.
+	#
+	# If you don't have these installed yet it's going to take a long time, but
+	# you'll only need to do it once.
+	pyenv install -s 3.6.2
+	pyenv install -s 3.5.3
+	pyenv install -s 3.4.4
+
+	# Set up the dev env and the environments for Tox
+	pyenv local ${MODULE_NAME} 3.6.2 3.5.3 3.4.4
```

Comment:
Isn't `pyenv local` restricted to a single local version?

Reply:
good question, this makefile comes from a temple public project, I think it's necessary for tox

Reply:
Oh hmm I missed this in my update to the template, testing on CircleCI doesn't use tox anymore (and nor would it locally unless we set that up again).

Reply:
I don't know if I'd put that effort in, tox was really challenging to get set up.

Reply:
Cool, probably worth updating this in the template to only install a recent Python version.